### PR TITLE
Tdr scripts backend

### PR DESCRIPTION
**Pull request by MancunianSam**

### CODE REVIEW

File: modules/environment-roles/templates/terraform_scripts_policy.json.tpl [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        "ssm:GetParameter"
+      ],
+      "Resource": [
+        "arn:aws:dynamodb:eu-west-2:${account_id}:table/tdr-terraform-state-lock-scripts",
+        "arn:aws:iam::${account_id}:instance-profile/bastion",
+        "arn:aws:iam::${account_id}:role/BastionEC2RoleIntg",
+        "arn:aws:kms:eu-west-2:${account_id}:alias/tdr-bastion-encryption-intg",
+        "arn:aws:kms:eu-west-2:${account_id}:key/*",
+        "arn:aws:s3:::tdr-terraform-state-scripts/*",
+        "arn:aws:ssm:eu-west-2:${account_id}:parameter/intg/consignmentapi/database/password",
```
Comment (TomJKing) [inline modules/environment-roles/templates/terraform_scripts_policy.json.tpl:76]: Should the TDR env be hardcoded: "... :parameter/**intg**/consignmentapi/..."

File: modules/environment-roles/templates/terraform_scripts_policy.json.tpl [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+        "kms:CreateKey",
+        "kms:ListAliases",
+        "kms:ListKeys",
+        "s3:ListObjects",
+        "sts:GetCallerIdentity"
+      ],
+      "Resource": "*"
+    }
+  ]
+}
```
Comment (TomJKing) [inline modules/environment-roles/templates/terraform_scripts_policy.json.tpl:119]: Missing line break

File: modules/permissions/variables.tf [hunk 3]
```diff
@@ -8,11 +8,21 @@ variable "terraform_state_bucket" {
...
+  type        = string
+}
+
 variable "terraform_state_lock" {
   description = "DynamoDb table controlling the terraform state lock"
   type        = string
 }
 
+variable "terraform_scripts_state_lock" {
+  description = "DynamoDb table controlling the tdr scripts terraform state lock"
```
Comment (TomJKing) [inline modules/permissions/variables.tf:22]: Capitalize "tdr"

File: modules/specific-environment-permissions/variables.tf [hunk 4]
```diff
@@ -41,4 +41,9 @@ variable "terraform_describe_account_arn" {
 variable "custodian_get_parameters_arn" {
   description = "IAM policy ARN for getting parameter values"
   type        = string
+}
+
+variable "terraform_scripts_state_bucket" {
+  description = "S3 bucket storing the terraform state for the TDR scripts terraform project"
+  type        = string
 }
```
Comment (TomJKing) [inline modules/specific-environment-permissions/variables.tf:49]: Missing line break

File: modules/state-lock/output.tf [hunk 5]
```diff
@@ -9,3 +9,7 @@ output "terraform_jenkins_state_lock" {
 output "terraform_grafana_state_lock_arn" {
   value = aws_dynamodb_table.terraform_state_lock_grafana.arn
 }
+
+output "terraform_scripts_state_lock_arn" {
+  value = aws_dynamodb_table.terraform_state_lock_scripts.arn
+}
```
Comment (TomJKing) [inline modules/state-lock/output.tf:15]: Missing line break

File: modules/state-lock/dynamodb.tf [hunk 6]
```diff
@@ -54,3 +54,22 @@ resource "aws_dynamodb_table" "terraform_state_lock_grafana" {
...
+    type = "S"
+  }
+
+  tags = merge(
+  var.common_tags,
+  map(
+  "Name", "TDR Scripts Terraform State Lock Table",
+  )
+  )
+}
```
Comment (TomJKing) [inline modules/state-lock/dynamodb.tf:75]: Missing line break

**Review by TomJKing — COMMENTED** [review]
Some minor comments. 

I think you need to run `terraform fmt` for some of the files in the PR.

File: modules/environment-roles/templates/terraform_scripts_policy.json.tpl (hunk 1 above)
Comment (MancunianSam) [reply modules/environment-roles/templates/terraform_scripts_policy.json.tpl:76]: No it definitely shouldn't be. I've passed in the environment instead.

File: modules/permissions/variables.tf (hunk 3 above)
Comment (MancunianSam) [reply modules/permissions/variables.tf]: Good point, I've changed it.

**Comment by MancunianSam** [comment]
I've added in the missing line breaks and I've ran terraform fmt over the whole thing.

**Review by TomJKing — APPROVED** [review]

**Pull request merged**